opty/tests/test_direct_collocation.py

```diff
@@ -1908,3 +1908,151 @@ def test_attributes_read_only():
+    par_map = {b1: 1.0,
+               b2: 2.0,
+               b3: 3.0
+               }
+    num_nodes = 26
+
+    # A: constant time interval
+    t0, tf = 0.0, 1.0
+    interval_value = tf/(num_nodes - 1)
+
+    def obj(free):
+        Fx = free[2*num_nodes:3*num_nodes]
+        return interval_value*np.sum(Fx**2)
+
+    def obj_grad(free):
+        grad = np.zeros_like(free)
+        l1 = 2*num_nodes
+        l2 = 3*num_nodes
+        grad[l1: l2] = 2.0*free[l1:l2]*interval_value
+        return grad
+
+    instance_constraints = (
+        x.func(t0),
+        ux.func(t0),
+        x.func(tf) - 1.0,
+        ux.func(tf),
+    )
+
+    bounds= {
+        a1: (-1.0, 1.0),
+        a2: (-1.0, -0.5),
+        a3: (-1.0, 1.0),
+
+        u1: (-1.0, 1.0),
+        u2: (-1.0, 1.0),
+        u3: (-1.0, 1.0),
+
+        x: (-1.0, 1.0 ),
+        ux: (-1.0, 1.0),
+        y: (-1.0, 1.0),
+        uy: (-1.0, 1.0),
+        z: (-1.0, 1.0),
+        uz: (-1.0, 1.0),
+    }
+
+    prob = Problem(
+        obj,
+        obj_grad,
+        eom,
+        state_symbols,
+        num_nodes,
+        interval_value,
+        known_parameter_map=par_map,
+        instance_constraints=instance_constraints,
+        bounds=bounds,
+        time_symbol=t,
+        )
+
+    initial_guess = np.zeros(prob.num_free)
+    start_idx = np.random.randint(0, num_nodes - 1)
+    for i in range(9):
+        initial_guess[start_idx + i*num_nodes] = 10.0
+    with warns(UserWarning):
+        _ , _ = prob.solve(initial_guess)
+
+    # B: variable time interval
+    h = sym.symbols('h')
+    interval_value = h
+    t0, tf = 0.0, (num_nodes - 1)*h
+
+    def obj(free):
+        Fx = free[6*num_nodes:9*num_nodes]
+        return free[-1]*np.sum(Fx**2)
+
+    def obj_grad(free):
+        grad = np.zeros_like(free)
+        l1 = 6*num_nodes
+        l2 = 9*num_nodes
+        grad[l1: l2] = 2.0*free[l1:l2]*free[-1]
+        grad[-1] = np.sum(free[l1:l2]**2)
+        return grad
+
+    instance_constraints = (
+        x.func(t0),
+        ux.func(t0),
+        x.func(tf) - 1.0,
+        ux.func(tf),
+    )
+    bounds= {
+        a1: (-1.0, -0.5),
+        a2: (-1.0, 1.0),
+        a3: (-1.0, 1.0),
+
+        u1: (-1.0, 1.0),
+        u2: (-1.0, 1.0),
+        u3: (-1.0, 1.0),
+
+        x: (-1.0, 1.0),
+        ux: (-1.0, 1.0),
+        y: (-1.0, 1.0),
+        uy: (-1.0, 1.0),
+        z: (-1.0, 1.0),
+        uz: (-1.0, 1.0),
+        h: (0.0, 0.4 )
+    }
+
+    prob = Problem(
+        obj,
+        obj_grad,
+        eom,
+        state_symbols,
+        num_nodes,
+        interval_value,
+        known_parameter_map=par_map,
+        instance_constraints=instance_constraints,
+        bounds=bounds,
+        time_symbol=t,
+        )
+
+    initial_guess = np.zeros(prob.num_free)
+    start_idx = np.random.randint(0, num_nodes - 1)
+    for i in range(9):
+        initial_guess[start_idx + i*num_nodes] = -10.0
+    initial_guess[-1] = 0.5
```

Comment:
You need a test that calls the new method explicitly as the primary test, i.e. does your new method do what it is supposed to do. This doesn't not require solving the problem.

Then additionally you need the test that `.solve()` does the right thing with and without the suggested flag above.

Reply:
Once you have the new `kwarg` you need to test if `True` and `False` do the right things.